pulseapi/entries/models.py

```diff
@@ -98,11 +119,22 @@ def thumbnail_image_tag(self):
     created = models.DateTimeField(
         default = timezone.now
     )
-    is_approved = models.BooleanField(
-        default=False,
-        help_text="Should this show up on the site?"
+
+    # moderation information
+    moderation_state = models.ForeignKey(
+        ModerationState,
+        related_name='entries',
+        default=get_default_moderation_state,
+        on_delete=models.PROTECT
     )
 
+    def set_moderation_state(self, state_name):
```

Comment:
I'm using this primarily to flip entries to "Approved", but this function will also be necessary for when the client has to do moderation calls. The reason it's doing `get_or_create` is mostly because that allows the code to just keep running instead of crashing when it encounters a new state it doesn't know about yet, and it's easy enough to go in and filter for unexpected moderation states in the admin view given that it auto-exposes all possible values as a filter.